telethon/network/mtprotosender.py

```diff
@@ -357,32 +358,36 @@ async def _reconnect(self, last_error):
         self._state.reset()
 
         retries = self._retries if self._auto_reconnect else 0
-        for attempt in retry_range(retries):
-            try:
-                await self._connect()
-            except (IOError, asyncio.TimeoutError) as e:
-                last_error = e
-                self._log.info('Failed reconnection attempt %d with %s',
-                               attempt, e.__class__.__name__)
+        if retries:
```

Comment:
Separating the range into its own method was exactly made to avoid this kind of extra works that needs to be done outside and keep the logic inside the method itself.

Reply:
Ok, now I get it maybe we can just take an extra parameter for `retry_range` for reconnecting and init the attempt before the loop.